backend/backend.c

```diff
@@ -0,0 +1,37 @@
+#include <wayland-server.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+#include "common/log.h"
+#include "backend.h"
+
+struct wlr_backend *wlr_backend_create(const struct wlr_backend_impl *impl,
+		struct wlr_backend_state *state) {
```

Comment:
Note `wlr_backend_autocreate` will eventually be made to automatically create the most appropriate backend. It will handle udev and logind and things like that, and will invoke the most appropriate backend's `wlr_*_backend_create` function. This function, `wlr_backend_create`, is an internal function used by the backends to create the user-facing backend object.